server/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -303,7 +303,7 @@ public Query doToQuery(SearchExecutionContext context) {
             luceneTopLeft.getLat(),
             luceneBottomRight.getLat()
         );
-        return geoShapeQueryable.geoShapeQuery(rectangle, fieldType.name(), SpatialStrategy.RECURSIVE, ShapeRelation.INTERSECTS, context);
+        return geoShapeQueryable.geoShapeQuery(context, fieldType.name(), SpatialStrategy.RECURSIVE, ShapeRelation.INTERSECTS, rectangle);
```

Comment:
Just call the new API directly.

Reply:
```suggestion
        return geoShapeQueryable.geoShapeQuery(context, fieldType.name(), ShapeRelation.INTERSECTS, rectangle);
```

Reply:
Needed for compatibility with `LegacyGeoShapeFieldMapper`